File: solution/kotlin/week06-exercise2/src/main/kotlin/report/CsvReportGenerator.kt
package report

import arrow.core.Either
import arrow.core.right

class CsvReportGenerator : CanGenerateReport {
    override fun generateReport(data: List<ReportData>): Either<ReportError, String> =
        buildString {
            appendLine("Starting CSV Report Generation...")
            appendLine("CSV Header: ID, Value, Description")
            data.forEach { d -> appendLine("${d.id},${d.value},${d.description}") }
            append("CSV Report Generated Successfully.")
        }.right()
}

File: solution/c#/week06-legacy-code/ReportGenerator/CsvReportGenerator.cs
using System.Text;
using LanguageExt;

namespace ReportGenerator
{
    public class CsvReportGenerator : ICanGenerateReport
    {
        public Either<ReportError, string> GenerateReport(Seq<ReportData> data)
        {
            var sb = new StringBuilder();
            sb.AppendLine("Starting CSV Report Generation...");
            sb.AppendLine("CSV Header: ID, Value, Description");
            foreach (var d in data)
            {
                sb.AppendLine($"{d.Id},{d.Value},{d.Description}");
            }

            sb.Append("CSV Report Generated Successfully.");
            return Prelude.Right<ReportError, string>(sb.ToString());
        }
    }
}